packages/material-ui/src/styles/experimentalStyled.test.js

```diff
@@ -500,5 +500,48 @@ describe('experimentalStyled', () => {
 
       expect(containsValidClass).to.equal(true);
     });
+
+    it('should not propagate classes props to component if it is a root slot', () => {
+      const Component = styled(
+        (props) => {
+          const { classes, ...other } = props;
+          return <div {...(classes && { 'data-with-classes': 'true' })} {...other} />;
+        },
+        {},
+        { name: 'MuiComponent', slot: 'Root' },
+      )`
+        width: 200px;
+        height: 300px;
+      `;
+
+      const { container } = render(<Component classes={{ root: 'foo' }}>Test</Component>);
+      expect(container.querySelector('[data-with-classes]')).to.equal(null);
+    });
+
+    it('should propagate classes props to component if it is not a root slot', () => {
+      const Component = styled(
+        (props) => {
+          const { classes, ...other } = props;
+          return <div {...(classes && { 'data-with-classes': 'true' })} {...other} />;
```

Comment:
```suggestion
          return <div data-with-classes={classes !== undefined} {...other} />;
```
relies on less syntax. And then use the same approach in both tests.

Reply:
Done 👍